### ISSUE TITLE
Fatal error when switching to context without any authorization

### ISSUE THREAD

**Issue by lstocchi** [redhat-developer/intellij-kubernetes#101]
While i was testing the extension i clicked on an old context. It resulted in the error below. It's super clear but the thing i didn't like was that it appeared as a `IDE Fatal Error`. Maybe a notification would be a better way to show it to the user??

```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://api.crc.testing:6443/api/v1/namespaces. Message: Unauthorized! Token may have expired! Please log-in again. Unauthorized.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:589)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:526)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:492)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:451)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:433)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:166)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:640)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:79)
	at org.jboss.tools.intellij.kubernetes.model.resource.NonNamespacedResourcesProvider.loadAllResources(NonNamespacedResourcesProvider.kt:36)
	at org.jboss.tools.intellij.kubernetes.model.resource.NonNamespacedResourcesProvider.getAllResources(NonNamespacedResourcesProvider.kt:28)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.getAllResources(ActiveContext.kt:132)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.access$getAllResources(ActiveContext.kt:58)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext$namespacedProviders$2.invoke(ActiveContext.kt:72)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext$namespacedProviders$2.invoke(ActiveContext.kt:58)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.getNamespacedProviders(ActiveContext.kt)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.watch(ActiveContext.kt:222)
	at org.jboss.tools.intellij.kubernetes.model.ResourceModel.watch(ResourceModel.kt:113)
	at org.jboss.tools.intellij.kubernetes.tree.TreeStructure$FolderDescriptor.watchResources(TreeStructure.kt:199)
	at org.jboss.tools.intellij.kubernetes.tree.ResourceWatchController$install$1.treeExpanded(ResourceWatchController.kt:28)
	at javax.swing.JTree.fireTreeExpanded(JTree.java:2775)
	at javax.swing.JTree.setExpandedState(JTree.java:3629)
	at javax.swing.JTree.expandPath(JTree.java:2212)
	at javax.swing.plaf.basic.BasicTreeUI.toggleExpandState(BasicTreeUI.java:2286)
	at javax.swing.plaf.basic.BasicTreeUI.handleExpandControlClick(BasicTreeUI.java:2273)
	at javax.swing.plaf.basic.BasicTreeUI.checkForClickInExpandControl(BasicTreeUI.java:2231)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.handleSelection(BasicTreeUI.java:3600)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.mousePressed(BasicTreeUI.java:3548)
	at com.intellij.util.ui.MouseEventAdapter.mousePressed(MouseEventAdapter.java:33)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.Component.processMouseEvent(Component.java:6545)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:409)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4523)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:664)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by adietish** [comment]
@lstocchi fully agree on this one, we should not have any non-harmful error pop up as IDE error. Furthermore this isn't a proper error, this is an expected inability to reach the given cluster/context. 
Can you elaborate on "notification", please? Do you have something specific in mind?

**Comment by lstocchi** [comment]
You could use the event log to show error/success messages or also notificatios popups. An example below. 

![image](https://user-images.githubusercontent.com/49404737/96562996-b59ebf80-12c1-11eb-9099-316ff8cb3a0e.png)

**Comment by adietish** [comment]
@lstocchi makes sense but then, I'm not aware of anything similar in IJ, but I didn't dig though. I'd think that for now logging as a warning or even an info is good enough?

**Comment by lstocchi** [comment]
That is a picture from IJ tekton. 
WDYM by warning/info log? Does it appear in the logs file or inside the terminal or somewhere else in the IDE?
If it's inside the logs file I'm not so convinced. In my case i was trying to switch cluster so i wish there had been a clear and visible message to know what wasn't working to immediately fix it. 
If the error is "hidden" inside the logs it's not so intuitive for me. Or atleast the plugin should show a notification "An error occurred during blabla check the logs for more infos". This way users are guided through the plugin and we prevent them from opening stupid issues.

**Comment by adietish** [comment]
This is duplicate to #124. In #124 fixes for #112 are already taken into account which leads to a slightly different behaviour. #124 is thus more accurate/up-to-date to track the issue.
Closing as Duplicate.
